cloudahoywriter.lua

```diff
@@ -81,13 +111,19 @@ local recOnG = 0.2
 local recOnB = 0.2
 local recOnA = 0.8
 
-local maybe_write_data -- forward declaration
+-- forward declarations
+local maybe_write_data 
+local automatic_recording_state_check
 
 -- Runs on every draw. May write to file. May show UI.
 function CAWR_on_every_draw()
+    automatic_recording_state_check()
     maybe_write_data()
 
-    if MOUSE_X > width * 3 then return end
+    if MOUSE_X > width * 3
+            and not should_show_recording_state_change() then
```

Comment:
Just a suggestion: maybe this could be called something like "should_force_ui_to_foreground"? (the recording_state_change could be left out as an implementation detail, maybe there will be other reasons why we force the ui)

Reply:
Good idea. Renamed the method.